language/move-prover/src/prover_task_runner.rs

```diff
@@ -32,20 +36,36 @@ pub trait ProverTask {
     fn init(&mut self, num_instances: usize) -> Vec<Self::TaskId>;
 
     /// Run the task with task_id. This function will be called from one of the worker threads.
-    async fn run(&mut self, task_id: Self::TaskId) -> Self::TaskResult;
+    async fn run(&mut self, task_id: Self::TaskId, sem: Arc<Semaphore>) -> Self::TaskResult;
+
+    /// Returns whether the task result is considered successful.
+    fn is_success(&self, task_result: &Self::TaskResult) -> bool;
 }
 
 pub struct ProverTaskRunner();
 
 impl ProverTaskRunner {
     /// Run `num_instances` instances of the prover `task` and returns the task id
     /// as well as the result of the fastest running instance.
-    pub fn run_tasks<T>(mut task: T, num_instances: usize) -> (T::TaskId, T::TaskResult)
+    pub fn run_tasks<T>(
+        mut task: T,
+        num_instances: usize,
+        sequential: bool,
+    ) -> (T::TaskId, T::TaskResult)
     where
         T: ProverTask + Clone + Send + 'static,
     {
-        let rt = Runtime::new().unwrap();
-
+        let rt = tokio::runtime::Builder::new()
+            .core_threads(2)
```

Comment:
Why do you have hardcoded 2 threads here? This restricts parallelism to two 2, no?

Should this be set to num_instances? Or perhaps removed and let Tokio decide how many threads to use based on available cores on the machine>